### ISSUE TITLE
Add in player ranks to class arez.MatchPlayer.

### ISSUE THREAD

**Issue by EthanHicks1** [DevilXD/HiRezAPI#1]
Although it would have to be manually processed for non-ranked matches (since this field would be empty/meaningless) it would be nice if it was possible to access rank data that's outside of a player object in a match so that the rank of a Private player can be accessed.

For example:
`ranked_keyboard`
and 
`ranked_controller`

to be accessible at the `arez.MatchPlayer` level instead of having to look in the `arez.Player` level.

**Comment by DevilXD** [comment]
As far as my fading knowledge goes, this information isn't accessible via the API, at least for the matches that have already been played out. The only place where this wrapper "skips" rank information, is in `arez.LivePlayer` (for live matches only), where it's only available for ranked queues (casual queues will just have `0` returned). The reason it's not there already, is because of this exact inconsistency, and if I were to add this, then it'd come with a giant red warning explaining this quirk. I can't do much more than this, as it's an API limitation that this information is available only in these specific circumstances.

Is this what you are asking about? If not, I could try asking about the possibility of adding rank information to matches that have already been played, but given the past responses on this topic being negative, I doubt it's going to go anywhere.

**Comment by EthanHicks1** [comment]
Yeah sorry, I overcomplicated that. It would just be to reveal that rank number, knowing that for most matches it will just be `0`.  As you mentioned, it would be up to the user to interpret this number.

**Comment by DevilXD** [comment]
Hmm. It appears that this is already a thing though: https://arez.readthedocs.io/en/latest/players.html#arez.LivePlayer.rank

Although without any warnings specifying it only works correctly for ranked matches, which I'm going to correct for the next update.

Again, I'm not sure if this request is for `arez.MatchPlayer` (past matches) or `arez.LivePlayer` (live matches) - if it's the former, I already explained that this information isn't provided by the API, and past discussions has shown that Hi-Rez wasn't keen on adding it back (it was a thing in the past). If it's the latter, as explained above, it's already there. So, which one is it?

**Comment by DevilXD** [comment]
As noted in a private discord message, the information about player's rank is indeed returned for already-played matches, but only for the ranked queues. The TP and amount of wins and losses is also returned, but not as important as the player's rank.

 This is currently not being parsed by the wrapper at all, but I'm willing to add the rank information, even if it's only available for ranked matches. You can expect it to become available in the next wrapper release.

**Comment by EthanHicks1** [comment]
I think this has been added to the main branch and this issue could be closed?

**Comment by DevilXD** [comment]
Nope, this still exists in the dev branch only. It's scheduled to release along with the upcoming 0.2.0, once I sort out one more problem, related to server status fetching from the API (currently, sometimes the API reports one of the servers being down, despite that being untrue, breaking the new, upcoming functionality). It's dependent entirely on Hi-Rez's ability to fix this at this point, as I continue to pass them data I collect on the issue - for more information, please see this Podio group thread (have to be a registered API user): https://podio.com/hirezstudioscom/smite-api-developer-collaboration/status/12521168

For now, if you'd be interested in the latest dev branch functionality, you can install it like so:
```
pip install --force-reinstall git+https://github.com/DevilXD/aRez@dev
```

**Comment by DevilXD** [comment]
This is now resolved for 0.2.0

https://arez.readthedocs.io/en/latest/players.html#arez.MatchPlayer.rank
https://arez.readthedocs.io/en/latest/players.html#arez.LivePlayer.rank
